lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/searcher/RemoteCentralRepositorySearcher.java

```diff
@@ -68,74 +99,95 @@ public Collection<Artifact> getArtifacts(Dependency artifactToSearch) {
+	public Collection<String> getPluginGroupIds(Dependency artifactToSearch) {
 		return disableCentralSearch ? Collections.emptySet() : internalGetGroupIds(artifactToSearch, PACKAGING_TYPE_MAVEN_PLUGIN);
 	}
 	
+	private static CacheManager<Request, Collection<Artifact>> artifactsCache = new CacheManager<>(cacheExpirationTimeoutMinutes);
 	private Collection<Artifact> internalGetArtifacts(Dependency artifactToSearch, String packaging) {
 		Request request = createArtifactIdsRequesty(artifactToSearch, packaging);
-		JsonObject responseBody = getResponseBody(request, artifactToSearch);
-		if (responseBody == null || responseBody.get("numFound").getAsInt() <= 0 ) {
-			return Collections.emptyList();
-		}
+		return artifactsCache.get(request, new Callable<Collection<Artifact>>() {
+			@Override
+			public Collection<Artifact> call() throws Exception {
+				JsonObject responseBody = getResponseBody(request, artifactToSearch);
+				if (responseBody == null || responseBody.get("numFound").getAsInt() <= 0 ) {
+					return Collections.emptyList();
+				}
 
-		List<Artifact> artifactInfos = new ArrayList<>();
-		responseBody.get("docs").getAsJsonArray().forEach(d -> {
-			artifactInfos.add(toArtifactInfo(d.getAsJsonObject()));
+				List<Artifact> artifactInfos = new ArrayList<>();
+				responseBody.get("docs").getAsJsonArray().forEach(d -> {
+					artifactInfos.add(toArtifactInfo(d.getAsJsonObject()));
+				});
+				
+				return artifactInfos;
+			}
 		});
-		return artifactInfos;
 	}
-	
-	private Set<ArtifactVersion> internalGetArtifactVersions(Dependency artifactToSearch, String packaging) {
+
+	private static CacheManager<Request, Collection<ArtifactVersion>> artifactVersionsCache = new CacheManager<>(cacheExpirationTimeoutMinutes);
+	private Collection<ArtifactVersion> internalGetArtifactVersions(Dependency artifactToSearch, String packaging) {
 		if (isEmpty(artifactToSearch.getArtifactId())) {
 			return Collections.emptySet();
 		}
 
 		Request request = createArtifactVersionsRequest(artifactToSearch, packaging);
-		JsonObject responseBody = getResponseBody(request, artifactToSearch);
-		if (responseBody == null || responseBody.get("numFound").getAsInt() <= 0 ) {
-			return Collections.emptySet();
-		}
-
-		Set<ArtifactVersion> artifactVersions = new HashSet<ArtifactVersion>();
-		responseBody.get("docs").getAsJsonArray().forEach(d -> {
-			artifactVersions.add(new DefaultArtifactVersion(d.getAsJsonObject().get(VERSION).getAsString()));
+		return artifactVersionsCache.get(request, new Callable<Collection<ArtifactVersion>>() {
+			@Override
+			public Collection<ArtifactVersion> call() throws Exception {
+				JsonObject responseBody = getResponseBody(request, artifactToSearch);
+				if (responseBody == null || responseBody.get("numFound").getAsInt() <= 0 ) {
+					return Collections.emptySet();
+				}
+	
+				Set<ArtifactVersion> artifactVersions = new HashSet<ArtifactVersion>();
+				responseBody.get("docs").getAsJsonArray().forEach(d -> {
```

Comment:
Use a DOMConstants or perhaps create a new DOMMavenRemoteCentralConstants to declare "docs" and "version"

I think it is better to have one DOM*constants per file type.

Reply:
I don't get why we should extract these constants into a separate class - `docs` and `numFound`  are internal names used only in Search Maven API response and not to be used outside of this class (unless it;s really required, of course). IMHO, no need to extract them into a separate class.

I can agree on converting them to  private static constants int the same class.

Reply:
The main problem when you extract some data from JSON is when you don't know the original structure it can be hard to understand what you are extracting. It th ereason why I suggest to declare all fields in a given class. You will see very quickly what fields you are extracting and if you give a sample of JSON it will help more.

Reply:
Having constants is nice when you want to retrieve where you are using for instance "docs" fields. Find references is your friend.

Reply:
I've added [JsonRemoteCentralRepositoryConstants](https://github.com/eclipse/lemminx-maven/blob/df3ee17bfa87af44ff70799c19beec44069ad2c0/lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/searcher/JsonRemoteCentralRepositoryConstants.java) class with JavaDoc comments and an example